src/NUnitFramework/framework/Attributes/ParallelizableAttribute.cs

```diff
@@ -51,6 +52,30 @@ public ParallelizableAttribute(ParallelScope scope) : base()
             Properties.Add(PropertyNames.ParallelScope, scope);
         }
 
+        /// <summary>
+        /// Overridden to check for invalid combinations of settings
+        /// </summary>
+        /// <param name="test"></param>
+        public override void ApplyToTest(Test test)
+        {
+            base.ApplyToTest(test);
+
+            if (test.RunState == RunState.NotRunnable)
+                return;
+
+            if (_scope.HasFlag(ParallelScope.Self) && _scope.HasFlag(ParallelScope.None))
+                MarkTestInvalid(test, "Test may not be both Parallel and NonParallel");
+
+            if (test is TestMethod && _scope.HasFlag(ParallelScope.ContextMask))
+                MarkTestInvalid(test, "ParallelScope of a test method may not specify Children or Fixtures");
+        }
+
+        private void MarkTestInvalid(Test test, string message)
```

Comment:
It could easily be a local extension method, used by this attribute only. Do you think that would be clear, or would it necessitate going to find the definition in order to understand it?

It couldn't be a general extension method within NUnit without violating slightly the independence of attributes. I've tried to keep most attribute definitions such that they could have been implemented as extensions because they then serve as a model for extensions and also because it forces us to exercise the extensibility mechanism all the time. The violation would only be slight, however, and the extension methods might be helpfull for those extending NUnit.